### ISSUE TITLE
Archlinux compilation and/or AUR package

### ISSUE THREAD

**Issue by LefterisJP** [occ-ai/obs-localvocal#62]
Hey thanks a lot for your work on this awesome package!

I wanted to try and build it under archlinux but the guide does not work due to 

```
   =>   Check Linux build requirements
   ✖︎   No apt-get command found. Please install apt
```

As archlinux uses pacman I guess some adjustment are needed but I am not very familiar with the best way to do it.

What would be amazing if this was packaged in an AUR package like someone did for your other extension to remove the background: https://github.com/Bitals/obs-backgroundremoval-aur/tree/master

Any help appreciated :heart:

**Comment by deweydbdecibel** [comment]
I absolutely love this idea!  I'm using Garuda and fish doesn't like me trying to install this either.  I spent hours trying to get brain to work on Garuda and ultimately frustration got the best of me.  :P  I'll be following this as I do a DJ stream and want to see if it will do captions for my ethnically diverse music selection!  

BTW, From what I see on your github, ya'll are doing some amazing work!

**Comment by royshil** [comment]
@umireon whats the process of getting an AUR on aur.archlinux.org ?

**Comment by umireon** [comment]
@royshil  I don't think downloading dependencies by CMake is not permitted on AUR so we need to change the build script to accept tarballs of dependencies first.

Anyway, here is the PKGBUILD script to build the package for pacman.

```
# Maintainer: Kaito Udagawa <umireon at gmail dot com>
pkgname=obs-localvocal

pkgver=0.1.0
pkgrel=1
pkgdesc='OBS plugin for local speech recognition and captioning using AI'
_source="${pkgname}-${pkgver}"

arch=(x86_64)
url='https://github.com/occ-ai/obs-localvocal'
license=('GPL2')
depends=('obs-studio' 'curl')
makedepends=('cmake' 'ninja')
source=("${_source}.tar.gz::$url/archive/refs/tags/${pkgver}.tar.gz")
sha256sums=('90b7321f2cc7e04854d05e97e9f9c6dcac7856923cc4465f0656049e110c6317')

build() {
  cd "$_source"
  cmake -B build --preset linux-x86_64 -DUSE_SYSTEM_CURL=ON
  cmake --build build
}

package() {
  cd "$_source"
  cmake --install build --prefix "$pkgdir/usr"
}
```
Gist: https://gist.github.com/umireon/eb83c2bd0657931345a3382bfbc4caed

And here are the instructions to build our plugin on Arch Linux

```
sudo pacman -Syu base-devel git
git clone https://gist.github.com/eb83c2bd0657931345a3382bfbc4caed.git obs-localvocal
makepkg -s
sudo pacman -U obs-localvocal-0.1.0-1-x86_64.pkg.tar.zst
```

Disclaimer: I haven't checked if this plugin works on OBS build in this way.

**Comment by LefterisJP** [comment]
So found some time to play with it.

Just by making the package manually with the `PKGBUILD` and `.SRCINFO` provided by @umireon it seems to succesfully build. For some reason ninja was not pulled as a build dependency but I manually installed it and it seems to have worked.

Now I have never used this plugin before so tried to use it but I am either doing something wrong or it's not properly installed.

![2024-02-05_12-57](https://github.com/occ-ai/obs-localvocal/assets/1658405/e66be6a9-7594-4b87-9441-9d3ec6685063)

As per the tutorial, I put it as a filter in the audio source and connected it to a text/subs element. But nothing appears there or in the logs.

**Comment by royshil** [comment]
@LefterisJP you get in the log "skipping inference" which means it can't hear any sound to process so it skips the Whisper inference

**Comment by LefterisJP** [comment]
> @LefterisJP you get in the log "skipping inference" which means it can't hear any sound to process so it skips the Whisper inference

yeah works now. Needed to speak louder and wait for quite some time to see the results. Need to play with the parameters I guess.

But anyway for the purpose of this issue the files to make the AUR package seem to work!

**Comment by LunasShade** [comment]
What is the status of the AUR package? ^^

**Comment by royshil** [comment]
is anyone willing to invest some time to make an AUR pkg?

**Comment by umireon** [comment]
We first need to organize dependent libraries to register our plugin to AUR. That's why I have not been tackling this request.

**Comment by umireon** [comment]
AUR won't allow the pre-built libraries, so we need to enable our plugin to build with the source of the dependent libraries.
If all the dependent libraries of localvocal can be provided from their source, I can quickly submit the PKGBUILD to AUR.

**Comment by royshil** [comment]
@suwagawaki looks like CTranslate2 is missing either as a shared or static library

**Comment by Seneral** [comment]
If anyone else gets this error:
```
CMake Error at cmake/common/helpers_common.cmake:89 (set_property):
  set_property can not be used on an ALIAS target.
Call Stack (most recent call first):
  CMakeLists.txt:26 (find_qt)
```
Then you need to manually apply this patch from the upstream OBS template:
https://github.com/obsproject/obs-plugintemplate/commit/a1289fdef404b08a7acbbf0d6d0f93da4c9fc087

@royshil sorry to ping you, was considering making a new issue instead. not sure what systems are affected, but since I'm also on an arch-based distro, I figured maybe it's related

**Comment by Seneral** [comment]
Additionally, the [install instructions in the readme for other distros](https://github.com/locaal-ai/obs-localvocal?tab=readme-ov-file#other-distros) are wrong, it uses `~/.config/obs-studio/plugins/data/` and `~/.config/obs-studio/plugins/bin/`, missing the obs-localvocal folder: `~/.config/obs-studio/plugins/obs-localvocal/xx`

**Comment by themobiusproject** [comment]
> ... missing the obs-localvocal folder: `~/.config/obs-studio/plugins/obs-localvocal/xx`

Thank you. This is where I got stuck as well. Here is a `tree` of the folder after it is installed and working.

```bash
$ tree ~/.config/obs-studio/plugins/obs-localvocal
/home/user/.config/obs-studio/plugins/obs-localvocal
├── bin
│   └── 64bit
│       ├── obs-localvocal
│       │   ├── libonnxruntime_providers_shared.so
│       │   ├── libonnxruntime.so -> libonnxruntime.so.1
│       │   ├── libonnxruntime.so.1 -> libonnxruntime.so.1.19.2
│       │   └── libonnxruntime.so.1.19.2
│       └── obs-localvocal.so
└── data
    ├── locale
    │   ├── ar-SA.ini
    │   ├── de-DE.ini
    │   ├── en-US.ini
    │   ├── es-ES.ini
    │   ├── fr-FR.ini
    │   ├── hi-IN.ini
    │   ├── ja-JP.ini
    │   ├── ko-KR.ini
    │   ├── pl-PL.ini
    │   ├── pt-BR.ini
    │   ├── ru-RU.ini
    │   └── zh-CN.ini
    └── models
        ├── ggml-model-whisper-tiny-en
        │   └── ggml-model-whisper-tiny.en.bin
        ├── models_directory.json
        └── silero-vad
            └── silero_vad.onnx
```

**Comment by hopeseekr** [comment]
I get stopped out on this:

```
-- Found WrapVulkanHeaders: /usr/include
CMake Error at cmake/common/helpers_common.cmake:89 (set_property):
  set_property can not be used on an ALIAS target.
Call Stack (most recent call first):
  CMakeLists.txt:27 (find_qt)
```

I can maintain the AUR package if i can successfully build it.

**Comment by Tabby** [comment]
I have merged a bunch of fixes to the CMake files so the build instructions for other distros definitely now work on Arch (I tested it in an Arch linux VM I set up specifically to make sure it worked)

> Additionally, the [install instructions in the readme for other distros](https://github.com/locaal-ai/obs-localvocal?tab=readme-ov-file#other-distros) are wrong, it uses `~/.config/obs-studio/plugins/data/` and `~/.config/obs-studio/plugins/bin/`, missing the obs-localvocal folder: `~/.config/obs-studio/plugins/obs-localvocal/xx`

My bad, thanks. Fixed now ^_^

**Comment by Tabby** [comment]
v0.5.1 now allows you to build on any distro with any or all of the available acceleration methods depending on what's installed on the system

I'm assuming this is fine for Arch's package management system but if you need the ability to actually toggle acceleration features on/off explicitly then please let me know

For Arch you'll probably want to build from source using CMake directly I suspect, rather than using the build scripts that use prebuilt dependencies, but they might work

Closing this as completed now that this is doable, but please do reopen it if you need to
